test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeConfiguration.cs

```diff
@@ -23,9 +23,127 @@ public class EdgeConfiguration
         readonly string deviceId;
         readonly IotHub iotHub;
 
-        public EdgeConfiguration(string deviceId, string agentImage, IotHub iotHub)
+        public class Module
         {
-            this.config = GetBaseConfig(agentImage);
+            readonly string name;
+            readonly EdgeConfiguration config;
+
+            public Module(string name, EdgeConfiguration config)
+            {
+                this.name = name;
+                this.config = config;
+            }
+
+            public Module WithEnvironment(IEnumerable<ValueTuple<string, string>> env)
```

Comment:
Feels like the builder pattern will fit nicely here.

Reply:
See my latest commits. They introduce a builder API for deployment config...